entity-framework/core/miscellaneous/cli/services.md

```diff
@@ -2,11 +2,51 @@
 
-[!code-csharp[Main](../../../../samples/core/Miscellaneous/CommandLine/DesignTimeServices.cs)]
+[!code-csharp[Main](../../../../samples/core/Miscellaneous/CommandLine/DesignTimeServices.cs#DesignTimeServices)]
+
+## Referencing Microsoft.EntityFrameworkCore.Design
+
+Microsoft.EntityFrameworkCore.Design is a DevelopmentDependency package. This means that the dependency won't flow transitively into other projects, and that you cannot, by default, reference its types.
+
+In order to reference its types and override design-time services, update the PackageReference item's metadata in your project file.
+
+```xml
+<PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="3.1.9">
+  <PrivateAssets>all</PrivateAssets>
+  <!-- Remove IncludeAssets to allow compiling against the assembly -->
+  <!--<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>-->
+</PackageReference>
+```
+
+If the package is being referenced transitively via Microsoft.EntityFrameworkCore.Tools, you will need to add an explicit PackageReference to the package and change its metadata.
+
+## List of services
+
+The following is a list of the design-time services.
+
+Service                                                                              | Description
+------------------------------------------------------------------------------------ | -----------
+<xref:Microsoft.EntityFrameworkCore.Design.IAnnotationCodeGenerator>                 | Generates the code for corresponding model annotations.
+<xref:Microsoft.EntityFrameworkCore.Design.ICSharpHelper>                            | Helps with generating C# code.
+<xref:Microsoft.EntityFrameworkCore.Design.IPluralizer>                              | Pluralizes and singularizes words.
+<xref:Microsoft.EntityFrameworkCore.Migrations.Design.IMigrationsCodeGenerator>      | Generates code for a migration.
+<xref:Microsoft.EntityFrameworkCore.Migrations.Design.IMigrationsScaffolder>         | The main class for managing migration files.
+<xref:Microsoft.EntityFrameworkCore.Scaffolding.IDatabaseModelFactory>               | Creates a database model from a database.
+<xref:Microsoft.EntityFrameworkCore.Scaffolding.IModelCodeGenerator>                 | Generates code for a model.
+<xref:Microsoft.EntityFrameworkCore.Scaffolding.IProviderConfigurationCodeGenerator> | Generates OnConfiguring code.
+<xref:Microsoft.EntityFrameworkCore.Scaffolding.IReverseEngineerScaffolder>          | The main class for scaffolding reverse engineered models.
+
+## Using services
+
+These services can also be useful for creating your own tools. For example, when you want to automate part of you design-time workflow.
+
+You can build a service provider containing these services using the AddEntityFrameworkDesignTimeServices and AddDbContextDesignTimeServices extension methods.
```

Comment:
Should we reference these methods to API?

Reply:
I don't think there's any value in doing that.